### ISSUE TITLE
EVP: cipher suite code hygiene

### ISSUE THREAD

**Issue by DDvO** [openssl/openssl#18443]
Having a look at the code in  #18430 and comparing it with the implementation of the existing cipher suites in `crypto/evp/e_*.c`,
I found quite some overlap among all of them, obviously created by copy&paste, in particular in the functions

```
static int XYZ_cipher(EVP_CIPHER_CTX *ctx, unsigned char *out, const unsigned char *in, size_t len)
static int XYZ_ctrl(EVP_CIPHER_CTX *ctx, int type, int arg, void *ptr)
```
for instance regarding hashing and padding.
Moreover, various meanwhile deprecated functions are used for HMAC calculations and the like.

Are there any plans to consolidate/clean up/update this codebase?

**Comment by levitte** [comment]
There is an intention to clean away everything legacy at some point.  At that time, ENGINE is finally going away.  Most probably, many deprecated symbols will disappear from public view.  At that time, we can most probably move a lot of code from `crypto/*` to `providers/implementations/...`

We haven't made any firm plans on this topic yet.
(I have a personal intention to make a configuration where all legacy code is disabled...  I just have to find some time)

**Comment by mattcaswell** [comment]
> in crypto/evp/c_allc.c have basically no effect on the availability of the ciphers, right?

They make the legacy function `EVP_get_cipherbyname` work. We should encourage people to use `EVP_CIPHER_fetch` instead - but I think there is still a lot of use of the older function.

We shouldn't be adding new ciphers to the legacy list. New ciphers should be added to the default provider.

**Comment by nhorman** [comment]
marking as inactive, to be closed at the end of 3.4 dev barring further input
